test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,366 @@
+            $specificationArray['media_type']
+        );
+
+        $specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+
+        $specificationConfig = $specificationLoader->get(self::SPECIFICATION_NAME);
+        Assert::assertEquals($expectedSpecificationConfig, $specificationConfig);
+    }
+
+    public function testLoadMoreThenOneSpecificationFilePathThrowsException(): void
+    {
+        $specificationFileName = 'test1';
+        $this->fileLocator->expects(self::once())
+            ->method('locate')
+            ->willReturn(['test1', 'test2']);
+        $specificationLoader = new SpecificationLoader(
+            $this->specificationParser,
+            $this->fileLocator,
+            $this->cache
+        );
+        $specificationArray  = $this->getSpecificationArray($specificationFileName);
+
+        $specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage(sprintf('More than one file path found for specification "%s".', __DIR__ . '/' . $specificationFileName));
+
+        $specificationLoader->load(self::SPECIFICATION_NAME);
+    }
+
+    public function testLoadNotLocalSpecificationFileThrowsException(): void
+    {
+        $specificationFileName = 'https://missed_specification.spec';
+        $this->fileLocator->expects(self::once())
+            ->method('locate')
+            ->willReturn($specificationFileName);
+        $specificationLoader = new SpecificationLoader(
+            $this->specificationParser,
+            $this->fileLocator,
+            $this->cache
+        );
+        $specificationArray  = $this->getSpecificationArray($specificationFileName);
+
+        $specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage(sprintf('This is not a local file "%s".', $specificationFileName));
+
+        $specificationLoader->load(self::SPECIFICATION_NAME);
+    }
+
+    public function testLoadMissedSpecificationFileThrowsException(): void
+    {
+        $specificationFileName = 'missed_specification.spec';
+        $this->fileLocator->expects(self::once())
+            ->method('locate')
+            ->willReturn($specificationFileName);
+        $specificationLoader = new SpecificationLoader(
+            $this->specificationParser,
+            $this->fileLocator,
+            $this->cache
+        );
+        $specificationArray  = $this->getSpecificationArray($specificationFileName);
+
+        $specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage(sprintf('File "%s" not found', $specificationFileName));
+
+        $specificationLoader->load(self::SPECIFICATION_NAME);
+    }
+
+    public function testLoadSavesSpecificationInCache(): void
```

Comment:
Missed a small case on the last review.
This test method should be run with a specification in `json` format too, to cover lines 122-12
```php
if ($type === 'json') {
    $specification = Reader::readFromJsonFile($specPath);
}
```